Src/MigrationExecutorFunctionApp/DocumentFeedMigrator.cs

```diff
@@ -0,0 +1,90 @@
+//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+namespace MigrationExecutorFunctionApp
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.CosmosDB.BulkExecutor;
+    using Microsoft.Azure.CosmosDB.BulkExecutor.BulkImport;
+    using Microsoft.Azure.Documents;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+
+    public class DocumentFeedMigrator
+    {
+        private IBulkExecutor bulkExecutor;
+
+        public DocumentFeedMigrator(IBulkExecutor bulkExecutor)
+        {
+            this.bulkExecutor = bulkExecutor;
+        }
+
+        [FunctionName("DocumentFeedMigrator")]
+        public async Task Run(
+            [Queue("%QueueName%", Connection = "QueueConnectionString")]IAsyncCollector<Document> postMortemQueue,
+            [CosmosDBTrigger(
+            databaseName: "%SourceDatabase%",
+            collectionName: "%SourceCollection%",
+            ConnectionStringSetting = "SourceCosmosDB",
+            LeaseCollectionName = "leases",
+            StartFromBeginning = true,
+            MaxItemsPerInvocation = 10000000,
+            CreateLeaseCollectionIfNotExists = true)
+            ]IReadOnlyList<Document> documents,
+            ILogger log)
+        {
+            if (documents != null && documents.Count > 0)
+            {
+                BulkImportResponse bulkImportResponse = null;
+
+                CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
```

Comment:
nit: You can delete this cancellationTokenSource and wherever you need one, just `default(CancellationToken)`